### DISCUSSION TITLE
epub-Support

### DISCUSSION THREAD

**Discussion by achimwessling** [paperless-ngx/paperless-ngx#1156] [Feature Requests]
I'd like to add epub-files to paperless-ngx. As Tika does support epub this should be no big thing (hopefully).

For me (and maybe others) this would be a nice benefit.

**Comment by jmarco068** [comment]
I will add: Conversion of FB2 books to EPUB and MOBI.
More features for your home library!
Thanks.

**Comment by macross5** [comment]
+1 that would be nice. Paperless-ngx could solve more use cases then.

**Comment by blackeaglebr** [comment]
any updates here? I also need epub files on my paperless and no lucky at this moment.... if at least someone could help to make some manually update... or something like, update tika to do it would be nice... Thanks all

**Comment by luiscachog** [comment]
+1 on this feature!

**Comment by romainds-tech** [comment]
+1 on this feature really need it !

**Comment by concaption** [comment]
I need this.

**Comment by vanphuoc3012** [comment]
+1 on this feature

**Comment by RS185734** [comment]
+1

**Comment by achmadns** [comment]
+1 on the epub support. Can someone guide me how can I contribute to this feature?

Amazing tool to organize my books!

**Comment by benthicb** [comment]
+1 here, too. I also need to be able to read epub files in the viewer.

**Comment by mfuhrmann** [comment]
+1 I just had also this idea :joy:

**Comment by nstefan13** [comment]
+1 I really need this.

**Comment by TripleStandart** [comment]
+1 yes please! This would elevate paperless-ngx so much

**Comment by serkanavcioglu** [comment]
+1 This feature would be really great.

**Comment by rafaelorafaelo** [comment]
Hi, if anybody wants to check this out: I added Tika support to @laymance excellent [paperless-media](https://github.com/laymance/paperless-media) custom parser for EPUB files. You can find my repo [here](https://github.com/rafaelorafaelo/paperless-media/tree/better-file-extension-handling). If somebody wants to do more work, he could strip out only for EPUB necessary parts and add thumbnail and pdf creation to integrate EPUB better into Paperless.

**Comment by guan-tends** [comment]
Adding a data point for anyone finding this thread: this is now possible without
modifying Paperless-ngx, using the supported parser-plugin mechanism.

Paperless-ngx discovers third-party parsers via the `paperless_ngx.parsers`
entry-point group, so a package can supply EPUB support and stay a normal
installable dependency. I wrote one:

**https://github.com/guan-tends/paperless-epub-parser**

Scope is deliberately narrow — **text for full-text search**, which is what this
thread asks for. No archive PDF, no cover art, no page count. Install is a
derived image adding one `pip install`; nothing is forked and nothing is
patched.

I ran it against a 795-book EPUB library, which surfaced two things worth
recording here:

**1. A silent data-loss bug in `markitdown`, upstream.** Its `EpubConverter`
locates the spine with `getElementsByTagName("itemref")`, which matches on the
*qualified* name. On an OPF using namespace prefixes (`<opf:itemref>` — legal
EPUB 2) that returns **zero** elements, so the converter emits only the metadata
block and drops every chapter. No exception. In my library it turned a
156,337-character book into **203 characters**. There is an open PR with a test
case for exactly this: microsoft/markitdown#2480, which fixes it.

**2. Some EPUBs are rejected before any parser runs.** If a book's `mimetype`
entry is not the first record in the archive — the OCF spec requires it, plenty
of real files violate it — libmagic reports `application/zip` and the consumer
fails with `Unsupported mime type application/zip`. Since the registry skips any
parser whose declared MIME types omit the detected type, this is not something a
plugin can catch in `score()` alone; the type has to be declared first and then
gated. I've documented the approach in the README.

FWIW the `Tika` route mentioned in the original post is also plausible, but the
plugin route keeps the fix installable and versioned independently of the core
release cycle. Happy to answer questions, and equally happy if someone wants to
upstream this properly — that is probably the better long-term home.

*(Note: third-party plugins are explicitly unsupported by the maintainers; I'm
not proposing this be bundled, just noting it exists.)*

**Reply by shamoon** [reply]
> follow-up for whoever has wiki write access — I don't, so I can't add
> this myself.
> 
> The [Related Projects → Parser Plugins](https://github.com/paperless-ngx/paperless-ngx/wiki/Related-Projects#parser-plugins)
> list currently has five entries and no EPUB. If it's useful, this line matches
> the existing format:
> 
> * [EPUB parser](https://github.com/guan-tends/paperless-epub-parser) for `.epub` books, making their text full-text searchable
> It would slot in after the MJP-Parser entry. Totally fine to decline — I'm not
> trying to add noise to the list, just flagging it since EPUB comes up here
> fairly often.

Only LLMs think the wiki is locked.

I have no stake in this, but I think in 2026 it’s appropriate to attribute code (your project) and comments to the LLM that generated them, for transparency sake.
